Dendrite crashes immediately after startup

<!--
All bug reports must provide the following background information
Text between <!-- and --​> marks will be invisible in the report.

IF YOUR ISSUE IS CONSIDERED A SECURITY VULNERABILITY THEN PLEASE STOP
AND DO NOT POST IT AS A GITHUB ISSUE! Please report the issue responsibly by
disclosing in private by email to security@matrix.org instead. For more details, please
see: https://www.matrix.org/security-disclosure-policy/
-->

### Background information
<!-- Please include versions of all software when known e.g database versions, docker versions, client versions -->
- **Dendrite version or git SHA**: 0.14.1+df748c5
- **SQLite3 or Postgres?**: Postgres
- **Running in Docker?**: Yes
- **`go version`**: Unknown
- **Client used (if applicable)**: n/a

### Description

- **What** is the problem: Dendrite crashes immediately after startup
- **Who** is affected: Every local user and everyone trying to contact a local user
- **How** is this bug manifesting: Dendrite crashes during startup because of a missing create event causing a panic. See logs at the End for details
- **When** did this first appear: After the Room v12 update

<!--
Examples of good descriptions:
- What: "I cannot log in, getting HTTP 500 responses"
- Who: "Clients on my server"
- How: "Errors in the logs saying 500 internal server error"
- When: "After upgrading to 0.3.0"

- What: "Dendrite ran out of memory"
- Who: "Server admin"
- How: "Lots of logs about device change updates"
- When: "After my server joined Matrix HQ"

Examples of bad descriptions:
- What: "Can't send messages"  - This is bad because it isn't specfic enough. Which endpoint isn't working and what is the response code? Does the message send but encryption fail?
- Who: "Me" - Who are you? Running the server or a user on a Dendrite server?
- How: "Can't send messages" - Same as "What".
- When: "1 day ago" - It's impossible to know what changed 1 day ago without further input.
-->

### Steps to reproduce
<!-- Please try reproducing this bug before submitting it. Issues which cannot be reproduced risk being closed. -->

- no idea, potentially old issue with the DB that dendrite ignored in the past?

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from server logs, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.

Alternatively, please send logs to @kegan:matrix.org, @s7evink:matrix.org or @devonh:one.ems.host
with a link to the respective Github issue, thanks!
-->
### Logs
```
time="2025-08-12T18:05:43.057216437Z" level=info msg="Dendrite version 0.14.1+df748c5"
time="2025-08-12T18:05:43.120909242Z" level=info msg="Starting nats-server" component=jetstream
time="2025-08-12T18:05:43.120991457Z" level=info msg="  Version:  2.11.7" component=jetstream
time="2025-08-12T18:05:43.121014550Z" level=info msg="  Git:      [df748c5]" component=jetstream
time="2025-08-12T18:05:43.121037503Z" level=info msg="  Name:     monolith" component=jetstream
time="2025-08-12T18:05:43.121062096Z" level=info msg="  Node:     2VDOOQDB" component=jetstream
time="2025-08-12T18:05:43.121088708Z" level=info msg="  ID:       NDAJQVOWO2WU7LTJRZUFGKKLCPCFHOANQCEFM3M3XRKNZ4SIY5XEUR4E" component=jetstream
time="2025-08-12T18:05:43.121143310Z" level=warning msg="Maximum payloads over 8.00 MB are generally discouraged and could lead to poor performance" component=jetstream
time="2025-08-12T18:05:43.121254222Z" level=info msg="Starting JetStream" component=jetstream
time="2025-08-12T18:05:43.122167213Z" level=info msg="    _ ___ _____ ___ _____ ___ ___   _   __  __" component=jetstream
time="2025-08-12T18:05:43.122226031Z" level=info msg=" _ | | __|_   _/ __|_   _| _ \\ __| /_\\ |  \\/  |" component=jetstream
time="2025-08-12T18:05:43.122253623Z" level=info msg="| || | _|  | | \\__ \\ | | |   / _| / _ \\| |\\/| |" component=jetstream
time="2025-08-12T18:05:43.122280231Z" level=info msg=" \\__/|___| |_| |___/ |_| |_|_\\___/_/ \\_\\_|  |_|" component=jetstream
time="2025-08-12T18:05:43.122304232Z" level=info component=jetstream
time="2025-08-12T18:05:43.122325930Z" level=info msg="         https://docs.nats.io/jetstream" component=jetstream
time="2025-08-12T18:05:43.122348382Z" level=info component=jetstream
time="2025-08-12T18:05:43.122372883Z" level=info msg="---------------- JETSTREAM ----------------" component=jetstream
time="2025-08-12T18:05:43.122404512Z" level=info msg="  Max Memory:      8.70 GB" component=jetstream
time="2025-08-12T18:05:43.122436308Z" level=info msg="  Max Storage:     5.25 TB" component=jetstream
time="2025-08-12T18:05:43.122510144Z" level=info msg="  Store Directory: \"jetstream\"" component=jetstream
time="2025-08-12T18:05:43.122554708Z" level=info msg="  API Level:       1" component=jetstream
time="2025-08-12T18:05:43.122592387Z" level=info msg=------------------------------------------- component=jetstream
time="2025-08-12T18:05:43.123281666Z" level=info msg="  Starting restore for stream '$G > DendriteInputDeviceListUpdate'" component=jetstream
time="2025-08-12T18:05:43.123729562Z" level=warning msg="Filestore [DendriteInputDeviceListUpdate] Stream state detected prior state, could not locate msg block 12" component=jetstream
time="2025-08-12T18:05:43.123781400Z" level=warning msg="Filestore [DendriteInputDeviceListUpdate] Recovering stream state from index errored: prior state file" component=jetstream
time="2025-08-12T18:05:43.124227475Z" level=info msg="  Restored 0 messages for stream '$G > DendriteInputDeviceListUpdate' in 1ms" component=jetstream
time="2025-08-12T18:05:43.124446848Z" level=info msg="  Starting restore for stream '$G > DendriteInputRoomEvent'" component=jetstream
time="2025-08-12T18:05:43.125125544Z" level=warning msg="Filestore [DendriteInputRoomEvent] Stream state outdated, last block has additional entries, will rebuild" component=jetstream
time="2025-08-12T18:05:43.125192677Z" level=warning msg="Filestore [DendriteInputRoomEvent] Recovering stream state from index errored: prior state file" component=jetstream
time="2025-08-12T18:05:43.128210412Z" level=info msg="  Restored 207 messages for stream '$G > DendriteInputRoomEvent' in 4ms" component=jetstream
time="2025-08-12T18:05:43.128485427Z" level=info msg="  Starting restore for stream '$G > DendriteInputSigningKeyUpdate'" component=jetstream
time="2025-08-12T18:05:43.129031556Z" level=warning msg="Filestore [DendriteInputSigningKeyUpdate] Stream state detected prior state, could not locate msg block 3" component=jetstream
time="2025-08-12T18:05:43.129093469Z" level=warning msg="Filestore [DendriteInputSigningKeyUpdate] Recovering stream state from index errored: prior state file" component=jetstream
time="2025-08-12T18:05:43.129447763Z" level=info msg="  Restored 0 messages for stream '$G > DendriteInputSigningKeyUpdate' in 1ms" component=jetstream
time="2025-08-12T18:05:43.129713057Z" level=info msg="  Starting restore for stream '$G > DendriteOutputAppserviceEvent'" component=jetstream
time="2025-08-12T18:05:43.130180461Z" level=warning msg="Filestore [DendriteOutputAppserviceEvent] Stream state detected prior state, could not locate msg block 17" component=jetstream
time="2025-08-12T18:05:43.130259931Z" level=warning msg="Filestore [DendriteOutputAppserviceEvent] Recovering stream state from index errored: prior state file" component=jetstream
time="2025-08-12T18:05:43.130739363Z" level=info msg="  Restored 0 messages for stream '$G > DendriteOutputAppserviceEvent' in 1ms" component=jetstream
time="2025-08-12T18:05:43.130988830Z" level=info msg="  Starting restore for stream '$G > DendriteOutputClientData'" component=jetstream
time="2025-08-12T18:05:43.131615089Z" level=info msg="  Restored 0 messages for stream '$G > DendriteOutputClientData' in 1ms" component=jetstream
time="2025-08-12T18:05:43.131857638Z" level=info msg="  Starting restore for stream '$G > DendriteOutputKeyChangeEvent'" component=jetstream
time="2025-08-12T18:05:43.132320653Z" level=warning msg="Filestore [DendriteOutputKeyChangeEvent] Stream state detected prior state, could not locate msg block 12" component=jetstream
time="2025-08-12T18:05:43.132378133Z" level=warning msg="Filestore [DendriteOutputKeyChangeEvent] Recovering stream state from index errored: prior state file" component=jetstream
time="2025-08-12T18:05:43.133320033Z" level=info msg="  Restored 31 messages for stream '$G > DendriteOutputKeyChangeEvent' in 1ms" component=jetstream
time="2025-08-12T18:05:43.133703487Z" level=info msg="  Starting restore for stream '$G > DendriteOutputNotificationData'" component=jetstream
time="2025-08-12T18:05:43.134231511Z" level=warning msg="Filestore [DendriteOutputNotificationData] Stream state detected prior state, could not locate msg block 3" component=jetstream
time="2025-08-12T18:05:43.134290319Z" level=warning msg="Filestore [DendriteOutputNotificationData] Recovering stream state from index errored: prior state file" component=jetstream
time="2025-08-12T18:05:43.134686778Z" level=info msg="  Restored 0 messages for stream '$G > DendriteOutputNotificationData' in 1ms" component=jetstream
time="2025-08-12T18:05:43.134897396Z" level=info msg="  Starting restore for stream '$G > DendriteOutputReceiptEvent'" component=jetstream
time="2025-08-12T18:05:43.135409589Z" level=warning msg="Filestore [DendriteOutputReceiptEvent] Stream state detected prior state, could not locate msg block 99" component=jetstream
time="2025-08-12T18:05:43.135478969Z" level=warning msg="Filestore [DendriteOutputReceiptEvent] Recovering stream state from index errored: prior state file" component=jetstream
time="2025-08-12T18:05:43.135855522Z" level=info msg="  Restored 0 messages for stream '$G > DendriteOutputReceiptEvent' in 1ms" component=jetstream
time="2025-08-12T18:05:43.136055766Z" level=info msg="  Starting restore for stream '$G > DendriteOutputRoomEvent'" component=jetstream
time="2025-08-12T18:05:43.136530815Z" level=warning msg="Filestore [DendriteOutputRoomEvent] Stream state detected prior state, could not locate msg block 156" component=jetstream
time="2025-08-12T18:05:43.136587034Z" level=warning msg="Filestore [DendriteOutputRoomEvent] Recovering stream state from index errored: prior state file" component=jetstream
time="2025-08-12T18:05:43.162698164Z" level=info msg="  Restored 8,644 messages for stream '$G > DendriteOutputRoomEvent' in 27ms" component=jetstream
time="2025-08-12T18:05:43.162940197Z" level=info msg="  Starting restore for stream '$G > DendriteOutputSendToDeviceEvent'" component=jetstream
time="2025-08-12T18:05:43.163521842Z" level=info msg="  Restored 0 messages for stream '$G > DendriteOutputSendToDeviceEvent' in 1ms" component=jetstream
time="2025-08-12T18:05:43.163663105Z" level=info msg="  Recovering 1 consumers for stream - '$G > DendriteInputDeviceListUpdate'" component=jetstream
time="2025-08-12T18:05:43.164332246Z" level=info msg="  Recovering 42 consumers for stream - '$G > DendriteInputRoomEvent'" component=jetstream
time="2025-08-12T18:05:43.179143898Z" level=info msg="  Recovering 1 consumers for stream - '$G > DendriteInputSigningKeyUpdate'" component=jetstream
time="2025-08-12T18:05:43.179549174Z" level=info msg="  Recovering 2 consumers for stream - '$G > DendriteOutputAppserviceEvent'" component=jetstream
time="2025-08-12T18:05:43.180046171Z" level=info msg="  Recovering 1 consumers for stream - '$G > DendriteOutputClientData'" component=jetstream
time="2025-08-12T18:05:43.180311136Z" level=info msg="  Recovering 2 consumers for stream - '$G > DendriteOutputKeyChangeEvent'" component=jetstream
time="2025-08-12T18:05:43.180847677Z" level=info msg="  Recovering 1 consumers for stream - '$G > DendriteOutputNotificationData'" component=jetstream
time="2025-08-12T18:05:43.181107264Z" level=info msg="  Recovering 3 consumers for stream - '$G > DendriteOutputReceiptEvent'" component=jetstream
time="2025-08-12T18:05:43.181745182Z" level=info msg="  Recovering 3 consumers for stream - '$G > DendriteOutputRoomEvent'" component=jetstream
time="2025-08-12T18:05:43.182411767Z" level=info msg="  Recovering 2 consumers for stream - '$G > DendriteOutputSendToDeviceEvent'" component=jetstream
time="2025-08-12T18:05:43.229860887Z" level=info msg="Server is ready" component=jetstream
time="2025-08-12T18:05:43.236680919Z" level=info msg="Loading server ACLs..."
time="2025-08-12T18:05:49.210985267Z" level=info msg="Finished loading server ACLs" acls=67 duration=5.974249291s
time="2025-08-12T18:05:49.696533815Z" level=info msg="Enabled perspective key fetcher" num_public_keys=2 server_name=matrix.org
time="2025-08-12T18:05:50.155865982Z" level=warning msg="entropia.de pushed us an event but \"entropia.de\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"entropia.de\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:50.312754205Z" level=info msg="Gap filled by /get_missing_events, injecting 20 new events" context=missing room_id="!ping-v11.1:maunium.net" txn_event="$xLQTkUhT5TKemTuKc0o4bz3xfrTCHZwx0m--GciwRm8" txn_prev_events="[$dW77IZEUJi0Ne4kzchb_5cyEmBdw61Gwkie0eG30Uec]"
time="2025-08-12T18:05:50.337039170Z" level=warning msg="entropia.de pushed us an event but \"c-base.org\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"c-base.org\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:50.398578793Z" level=info msg="Gap filled by /get_missing_events, injecting 20 new events" context=missing room_id="!YTvKGNlinIzlkMTVRl:matrix.org" txn_event="$Rq6q7IzJ8JCgSImg8CZcdXvBRnVSZ3-yx3X0yOpsz6k" txn_prev_events="[$kY_iuuD73nqhmf9PstayHOVMv2l-rUqUObszU65UZlA $HTSp9-FEs1LBa0Tm_3qvFm1k152Cy5NauJEjfQuWffU $zYJWZGhBCO1DTRt3gfyc5_MOQmNg_21w6j3pQ6T6lpg $chsa6fc93A4hiYUEZrNRJXvZvHAB3_TiPNZYuxGfw8k]"
time="2025-08-12T18:05:50.498098014Z" level=warning msg="Stored rejected event" context=missing error="event \"$y0aClSXQmPUnXImRPivGRfcfvGRm5qiV9hhWl6ianEE\" has missing prev events" event_id="$y0aClSXQmPUnXImRPivGRfcfvGRm5qiV9hhWl6ianEE" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=org.matrix.dummy_event
time="2025-08-12T18:05:50.580293551Z" level=info msg="Gap filled by /get_missing_events, injecting 16 new events" context=missing room_id="!UbCmIlGTHNIgIRZcpt:nheko.im" txn_event="$ralfjgfBPJhFDWK_5IhhsTOb5z6vPBFOZHauS8e_wDw" txn_prev_events="[$ucRawlw29BWsFnaBdRoqgWfRFA9dLEzOiBmVuVuDfoE $qFmRYDjDe6wm-KiIRXn0cQ-wMLZ2xXPO8M0VZqWY6qE $qHMAlkltP-QQtRh9__-LY7UuK0yv9aaWg9RmE_BIRh0 $TeLqjYZuf8fgHAsm2q86N1-TE2d0sNv1Cd1fwZRNSWg $SDNicxfq22H3p3Dyw-I-Z1XfbWuHKUhNxR8phn9B9mE $QgNEknN3dzlmNTf9IXgLP1x5nslDzyZYjhg3f-LQFB0]"
time="2025-08-12T18:05:50.623940465Z" level=warning msg="entropia.de pushed us an event but \"d12macera.com\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"d12macera.com\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:50.820643908Z" level=info msg="Gap filled by /get_missing_events, injecting 19 new events" context=missing room_id="!itCOeHerJPucwbZJDz:element.io" txn_event="$JLUwQ0FVJn8ed3qUvTlT01alR_R2cw1huf4YK-cHMjc" txn_prev_events="[$FbTnPwzQwYxVP0qiTWDaIdTgJn39GEl-esXBonCYs58]"
time="2025-08-12T18:05:50.864040135Z" level=warning msg="Stored rejected event" context=missing error="event \"$o_zioNOi4iA9otSS1QBLASN1Fsg2kbXn5SoMBVB8Fuc\" has missing prev events" event_id="$o_zioNOi4iA9otSS1QBLASN1Fsg2kbXn5SoMBVB8Fuc" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:50.974304335Z" level=info msg="Gap filled by /get_missing_events, injecting 18 new events" context=missing room_id="!YQErrczcFwrcuZheEB:sw1v.org" txn_event="$n9i3o6rWUN4WschT8gL3GzVNv4J-5eRXbw-aDQ8tbB4" txn_prev_events="[$f9bbrwAJKF0USB_z-tcqA4dfhiLuBr88MzTH3tOcPOA]"
time="2025-08-12T18:05:51.010368640Z" level=warning msg="entropia.de pushed us an event but \"is-a-cutie.de\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"is-a-cutie.de\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:51.053996876Z" level=warning msg="Stored rejected event" context=missing error="event \"$7z2U4KDLEMtQVPwYR2cc6uKwvu71MvHNlhDQIuYPgMM\" has missing prev events" event_id="$7z2U4KDLEMtQVPwYR2cc6uKwvu71MvHNlhDQIuYPgMM" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:51.232147643Z" level=warning msg="entropia.de pushed us an event but \"mart-w.de\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"mart-w.de\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:51.316266736Z" level=warning msg="Stored rejected event" context=missing error="event \"$mNYLTOFq65DdjI-Fio3HbxLQn7RAGbu6zIkH-Jhq34c\" has missing prev events" event_id="$mNYLTOFq65DdjI-Fio3HbxLQn7RAGbu6zIkH-Jhq34c" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:51.391742505Z" level=warning msg="entropia.de pushed us an event but \"solarpunk.social\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 77 95 70 79 82 66 73 68 68 69 78 58 32 83 101 114 118 101 114 32 105 115 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"solarpunk.social\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: M_FORBIDDEN: Server is not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:51.478500360Z" level=warning msg="Stored rejected event" context=missing error="event \"$thv4pbzD3WnzkugKkf1MFzALZmjDFJ6tNeRa8VtpqCY\" has missing prev events" event_id="$thv4pbzD3WnzkugKkf1MFzALZmjDFJ6tNeRa8VtpqCY" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:51.675613061Z" level=warning msg="entropia.de pushed us an event but \"systemli.org\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"systemli.org\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:51.751056284Z" level=warning msg="Stored rejected event" context=missing error="event \"$9Y5zkM5zpmQmXE1iKwINGmxMOzuLualPAOXEFCAJAdE\" has missing prev events" event_id="$9Y5zkM5zpmQmXE1iKwINGmxMOzuLualPAOXEFCAJAdE" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:51.989206824Z" level=warning msg="Stored rejected event" context=missing error="event \"$4A2zuUV86ZatfoJorJQ9a3jEdRkBK0-gGxt8RlQQtqA\" has missing prev events" event_id="$4A2zuUV86ZatfoJorJQ9a3jEdRkBK0-gGxt8RlQQtqA" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:52.034291355Z" level=warning msg="entropia.de pushed us an event but \"zimward.moe\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"zimward.moe\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:52.143212808Z" level=warning msg="Stored rejected event" context=missing error="event \"$stOd2CcZIIU5zW0EFM-tE4rbyLedrVDdBbGyUxaHUEE\" has missing prev events" event_id="$stOd2CcZIIU5zW0EFM-tE4rbyLedrVDdBbGyUxaHUEE" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:52.254385058Z" level=warning msg="Stored rejected event" context=missing error="event \"$-ut-BlGf1dSEcl2HChF4h63kQ9Ub-ruMtHXgBDrBNts\" has missing prev events" event_id="$-ut-BlGf1dSEcl2HChF4h63kQ9Ub-ruMtHXgBDrBNts" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:52.488486137Z" level=warning msg="entropia.de pushed us an event but \"fairydust.space\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"fairydust.space\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:52.549598958Z" level=warning msg="Stored rejected event" context=missing error="event \"$zCy3d8yYu6FMg2Iww0ShmQDlN46K-SHXiPCjwETcIaY\" has missing prev events" event_id="$zCy3d8yYu6FMg2Iww0ShmQDlN46K-SHXiPCjwETcIaY" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:52.620825376Z" level=info msg=PerformDeviceCreation context=missing device_id=0xc0002e2ae0 display_name=0xc0002e2ae0 localpart=aL1FTNamgceHbXzamcUwpOcoJLWmASOt
time="2025-08-12T18:05:52.683507502Z" level=info msg=PerformDeviceCreation context=missing device_id=0xc004ab6d60 display_name=0xc004ab6d60 localpart=FWFUBLCauOZhmAlIrCDQO80L5pgXu9822UhrbUWbONfHU33uFqegRn3hFY46H-u5
time="2025-08-12T18:05:52.688663592Z" level=warning msg="Stored rejected event" context=missing error="event \"$vVlKkAGP3RAVCXcEyvzPrC25aPuirHDpSoY3k_aeQvc\" has missing prev events" event_id="$vVlKkAGP3RAVCXcEyvzPrC25aPuirHDpSoY3k_aeQvc" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:52.731860283Z" level=info msg="Enabling shared secret registration at /_synapse/admin/v1/register"
time="2025-08-12T18:05:52.756098996Z" level=warning msg="entropia.de pushed us an event but \"matrix.minionflo.net\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"matrix.minionflo.net\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:52.811770742Z" level=warning msg="entropia.de pushed us an event but \"matrix.org\" did not respond to /get_missing_events" context=missing error="contents=[123 34 101 114 114 99 111 100 101 34 58 34 77 95 70 79 82 66 73 68 68 69 78 34 44 34 101 114 114 111 114 34 58 34 72 111 115 116 32 110 111 116 32 105 110 32 114 111 111 109 46 34 125] msg=Failed to POST JSON (hostname \"matrix.org\" path \"/_matrix/federation/v1/get_missing_events/!PGLaiDpuXlGMmVAgif:nix2twink.gay\") code=403 wrapped=M_FORBIDDEN: Host not in room." event_id="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" room_id="!PGLaiDpuXlGMmVAgif:nix2twink.gay" txn_event="$686DOWbcjzJC6oB_FWQcgfqzfRQkzYmU5mUqLCBQ-PQ" txn_prev_events="[$MGH2tgfp8ZXYM-u1XRaG8GJp5khy6Tw5Z0klJUOkdtI]"
time="2025-08-12T18:05:52.887401765Z" level=warning msg="Stored rejected event" context=missing error="event \"$ytTWjBcTRKtmlGT5nauXh7KIFpSYzqU9cbXtx5lgFvo\" has missing prev events" event_id="$ytTWjBcTRKtmlGT5nauXh7KIFpSYzqU9cbXtx5lgFvo" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:53.055600090Z" level=warning msg="Stored rejected event" context=missing error="event \"$zt5VV_V5Aw8ljiKuVFj3L4CzSEVf_KxZGsfouzz7UJ0\" has missing prev events" event_id="$zt5VV_V5Aw8ljiKuVFj3L4CzSEVf_KxZGsfouzz7UJ0" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:53.116155927Z" level=warning msg="Stored rejected event" context=missing error="event \"$CKE3VpZw4Aq82mPGWrZhdB0w9wWAmJrz_i2YMR8LFzs\" has missing prev events" event_id="$CKE3VpZw4Aq82mPGWrZhdB0w9wWAmJrz_i2YMR8LFzs" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:53.251685842Z" level=warning msg="Stored rejected event" context=missing error="event \"$wVTVGg7G1-nryxR7fwWmPEoLKYjxYlZkiwkAwuMAUX4\" has missing prev events" event_id="$wVTVGg7G1-nryxR7fwWmPEoLKYjxYlZkiwkAwuMAUX4" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:53.442430003Z" level=warning msg="Stored rejected event" context=missing error="event \"$EA423kHWDQcrwXpvAAJeIDC-WcflHKy5wjMXdN2uPzc\" has missing prev events" event_id="$EA423kHWDQcrwXpvAAJeIDC-WcflHKy5wjMXdN2uPzc" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:53.722898886Z" level=warning msg="Stored rejected event" context=missing error="event \"$x5P2m4G_6zVS3pISQf8OWVTGP1II9O30xobsY8c07dI\" has missing prev events" event_id="$x5P2m4G_6zVS3pISQf8OWVTGP1II9O30xobsY8c07dI" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
time="2025-08-12T18:05:53.961289812Z" level=warning msg="Stored rejected event" context=missing error="event \"$64zuKvkZG1Pa__lSMVn5FD_1OytYbtXPNcfZsy6XHyM\" has missing prev events" event_id="$64zuKvkZG1Pa__lSMVn5FD_1OytYbtXPNcfZsy6XHyM" kind=KindOld origin=usbpc.xyz room_id="!ping-v11.1:maunium.net" type=m.room.message
panic: ResolveStateConflictsV2: cannot find create event in auth chain

goroutine 398 [running]:
github.com/matrix-org/gomatrixserverlib.ResolveStateConflictsV2({0xc00693a488, 0x22, 0x47}, {0xc009014000, 0xb7e, 0xe00}, {0x2a12840, 0x0, 0x0}, 0xc003cbf700, ...)
	github.com/matrix-org/gomatrixserverlib@v0.0.0-20250811193806-b7e0e0824751/stateresolutionv2.go:77 +0x15a8
github.com/matrix-org/gomatrixserverlib.ResolveConflicts({0x1a5466a, 0x2}, {0xc0086c0000, 0x1643, 0x7f0dd2cd5be0?}, {0x2a12840, 0x0, 0x0}, 0xc003cbf700, 0xc003cbf720)
	github.com/matrix-org/gomatrixserverlib@v0.0.0-20250811193806-b7e0e0824751/stateresolution.go:398 +0x6d6
github.com/element-hq/dendrite/roomserver/internal/input.(*missingStateReq).resolveStatesAndCheck(0xc00683e420, {0x1d5a660?, 0xc0061467e0?}, {0x1a5466a, 0x2}, {0xc00679d520, 0x2, 0x0?}, {0x1d741a0, 0xc006d568c0})
	github.com/element-hq/dendrite/roomserver/internal/input/input_missing.go:497 +0x509
github.com/element-hq/dendrite/roomserver/internal/input.(*missingStateReq).lookupResolvedStateBeforeEvent(0xc00683e420, {0x1d5a660?, 0xc00052d200?}, {0x1d741a0, 0xc006d568c0}, {0x1a5466a, 0x2})
	github.com/element-hq/dendrite/roomserver/internal/input/input_missing.go:320 +0x8a5
github.com/element-hq/dendrite/roomserver/internal/input.(*missingStateReq).processEventWithMissingState(0xc00683e420, {0x1d5a660?, 0xc007a67b30?}, {0x1d741a0, 0xc0076c7b80}, {0x1a5466a, 0x2})
	github.com/element-hq/dendrite/roomserver/internal/input/input_missing.go:138 +0x678
github.com/element-hq/dendrite/roomserver/internal/input.(*Inputer).processRoomEvent(0xc007926100, {0x1d5a660, 0xc007a67b00}, {0xc0076f25ae, 0xa}, 0xc0004dd3e0)
	github.com/element-hq/dendrite/roomserver/internal/input/input_events.go:258 +0x1bfe
github.com/element-hq/dendrite/roomserver/internal/input.(*worker)._next(0xc00792e320)
	github.com/element-hq/dendrite/roomserver/internal/input/input.go:334 +0x545
github.com/Arceliar/phony.(*Inbox).run(0xc00792e320)
	github.com/Arceliar/phony@v0.0.0-20220903101357-530938a4b13d/actor.go:98 +0x2d
created by github.com/Arceliar/phony.(*Inbox).restart in goroutine 391
	github.com/Arceliar/phony@v0.0.0-20220903101357-530938a4b13d/actor.go:132 +0x4f
```

--- a/go.mod
+++ b/go.mod
@@ -25,7 +25,7 @@ require (
 	github.com/matrix-org/dugong v0.0.0-20210921133753-66e6b1c67e2e
 	github.com/matrix-org/go-sqlite3-js v0.0.0-20220419092513-28aa791a1c91
 	github.com/matrix-org/gomatrix v0.0.0-20220926102614-ceba4d9f7530
-	github.com/matrix-org/gomatrixserverlib v0.0.0-20250814102638-60b9d3e5b634
+	github.com/matrix-org/gomatrixserverlib v0.0.0-20250815065806-6697d93cbcba
 	github.com/matrix-org/pinecone v0.11.1-0.20230810010612-ea4c33717fd7
 	github.com/matrix-org/util v0.0.0-20221111132719-399730281e66
 	github.com/mattn/go-sqlite3 v1.14.28

--- a/go.mod
+++ b/go.mod
@@ -25,7 +25,7 @@ require (
 	github.com/matrix-org/dugong v0.0.0-20210921133753-66e6b1c67e2e
 	github.com/matrix-org/go-sqlite3-js v0.0.0-20220419092513-28aa791a1c91
 	github.com/matrix-org/gomatrix v0.0.0-20220926102614-ceba4d9f7530
-	github.com/matrix-org/gomatrixserverlib v0.0.0-20250813150445-9f5070a65744
+	github.com/matrix-org/gomatrixserverlib v0.0.0-20250814102638-60b9d3e5b634
 	github.com/matrix-org/pinecone v0.11.1-0.20230810010612-ea4c33717fd7
 	github.com/matrix-org/util v0.0.0-20221111132719-399730281e66
 	github.com/mattn/go-sqlite3 v1.14.28

